src/openforms/registrations/contrib/zgw_apis/options.py

```diff
@@ -140,81 +154,163 @@ def validate(self, attrs: dict[str, Any]) -> dict[str, Any]:
             else:
                 attrs["zgw_api_group"] = existing_group
 
-        group_config = attrs["zgw_api_group"]
+        _validate_against_catalogi_api(attrs)
 
-        # Run all validations against catalogi API in the same connection pool.
-        with get_catalogi_client(group_config) as client:
-            catalogi = list(client.get_all_catalogi())
+        return attrs
 
-            # validate that the zaaktype is in the provided catalogi
-            zaaktype_url = attrs["zaaktype"]
-            zaaktype_exists = any(
-                zaaktype_url in catalogus["zaaktypen"] for catalogus in catalogi
-            )
-            if not zaaktype_exists:
-                raise serializers.ValidationError(
-                    {
-                        "zaaktype": _(
-                            "The provided zaaktype does not exist in the specified "
-                            "Catalogi API."
-                        )
-                    },
-                    code="not-found",
-                )
 
-            # validate that the informatieobjecttype is in the provided catalogi
-            informatieobjecttype_url = attrs["informatieobjecttype"]
-            informatieobjecttype_exists = any(
-                informatieobjecttype_url in catalogus["informatieobjecttypen"]
-                for catalogus in catalogi
+def _validate_against_catalogi_api(attrs: RegistrationOptions) -> None:
```

Comment:
in my mind, `validators.py` is more intended for re-usable validators, and this validation helper is tied very strongly to the particular serializer that it's pretty much impossible to re-use.

These functions basically only exist to make the code a bit more readable, this could as well just have been directly implemented in `def valiate(self, attrs):` entirely, but that became such a huge chunk of code that I found it not maintainable anymore.

I also dislike the indirection created by moving private helper functions into different files, which requires me to click around more to see what the shape of data is actually like.

Then finally, `validators.py` are aimed at the django model layer, while these helpers are intended for DRF serializers. Unfortunately, these validators are not interchangeable (I tried building a generic solution but it gets real messy real quick). So, `validators.py` is used in the admin, while the serializer is used in the API endpoints used by the form designer.

Reply:
Understood and yes after your explanation makes sense.